pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/XMLParser.kt

```diff
@@ -10,59 +10,96 @@ import javax.xml.parsers.DocumentBuilderFactory
-                val indexes = (0 until indexesNodeList.length).map { indexesNodeList.item(it).textContent.toInt() }
-                val blocksNodeList = element.getElementsByTagName("block")
-                val blocks = (0 until blocksNodeList.length).map { blocksNodeList.item(it).textContent.toInt() }
-                val killingTest = element.getElementsByTagName("killingTest").item(0).textContent
-                val description = element.getElementsByTagName("description").item(0).textContent
+                    val detected = getAttribute(element, "detected", false)
+                    val status = getAttribute(element, "status", "N/A")
+                    val numberOfTestsRun = getAttribute(element, "numberOfTestsRun", -1)
+                    val sourceFile = getTextContent(element, "sourceFile")
+                    val mutatedClass = getTextContent(element, "mutatedClass")
+                    val mutatedMethod = getTextContent(element, "mutatedMethod")
+                    val methodDescription = getTextContent(element, "methodDescription")
+                    val lineNumber = getAttribute(element, "lineNumber", -1)
+                    val mutator = getTextContent(element, "mutator")
+                    val indexes = getListContent(element, "index")
+                    val blocks = getListContent(element, "block")
+                    val killingTest = getTextContent(element, "killingTest")
+                    val description = getTextContent(element, "description")
 
-                // Create a MutationResult object and add it to the data structure
-                val mutationResult = MutationResult(
-                    detected,
-                    status,
-                    numberOfTestsRun,
-                    sourceFile,
-                    mutatedClass,
-                    mutatedMethod,
-                    methodDescription,
-                    lineNumber,
-                    mutator,
-                    indexes,
-                    blocks,
-                    killingTest,
-                    description
-                )
-                resultData.addMutationResult(mutationResult)
+                    val mutationResult = MutationResult(
+                        detected,
+                        status,
+                        numberOfTestsRun,
+                        sourceFile,
+                        mutatedClass,
+                        mutatedMethod,
+                        methodDescription,
+                        lineNumber,
+                        mutator,
+                        indexes,
+                        blocks,
+                        killingTest,
+                        description
+                    )
+                    resultData.addMutationResult(mutationResult)
+                }
             }
+        } catch (e: Exception) {
+            // Handle parsing errors
+            e.printStackTrace()
```

Comment:
a TODO comment would probably be good here to show that the error handline is not done